### ISSUE TITLE
[Android] FileStream.openAsync() doesn't complete when using a content URI

### ISSUE THREAD

**Issue by ajwfrost** [airsdk/Adobe-Runtime-Support#2771]
Using recent AIR SDKs with support for Android 'content://' based URIs, if you use such a file in a call to `FileStream.openAsync(f, FileMode.READ)` then the `complete` event is never fired. The same File object works fine and can be read when using the synchronous `FileStream.open()` method.

This was reported to us via email - adding details here in case anyone has seen this/similar issues. Will be fixed shortly...

thanks

**Comment by andrefmferreira** [comment]
Hi everyone,

i've tested the sdk published last August's 19th, but the result is the same.  the code i have:

```
private function onFileSelected(event:Event):void {
  trace("File selected");
  try {
	  var file: File = event.target as File;

	  excelFileStream = new FileStream();
	  excelFileStream.addEventListener(Event.COMPLETE, onFileLoaded);
	  excelFileStream.openAsync(file, FileMode.READ);
	  loadingIndicator.visible = false;
  
  } catch (error:Error) {
	  trace("ERRO: ", error.message);
  }
  
}
```

When a file is selected, the stream should open the file, and when completed, the onFileLoaded should be triggered. This last step is not happening.

If i try the open method, it will trigger an exception, saying the file or directory does not exist. but it does:
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/16455146/4f479977-e387-4f5b-87a9-6677aae2cb73)


Any thoughts?

Thanks ;)
André Ferreira

**Comment by ajwfrost** [comment]
Hi

It seems there were a couple of issues.. we'd fixed one issue, but the actual opening of a file with that style of path was going wrong (a document URI, rather than a tree URI.. Android seems to be very fluid in what style of URI it actually returns from some of these APIs!) - and we've then found another bug that was causing an exception when trying to read a file over 8k..

Can you try extracting the .jar file from the below zip, and copying this over the one in the AIR SDK lib/android/lib folder? This hopefully addresses all the problems..

thanks

   Andrew

[runtimeClasses.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/12426417/runtimeClasses.zip)

**Comment by andrefmferreira** [comment]
Hi @ajwfrost ,

Works like a charme 👍 

Many thanks!!!!! 
You are the best!!

Regards,
André

**Comment by ajwfrost** [comment]
Thanks for confirming -> we'll get those changes out in our next release but feel free to use that patch file for now..
thanks

**Comment by marchbold** [comment]
Resolved in [50.2.3.4](https://airsdk.dev/news/2023/08/19/air-release)
